### DISCUSSION TITLE
custom nodes / extensions overview?

### DISCUSSION THREAD

**Discussion by cool-bigdogs-tshirt** [Comfy-Org/ComfyUI#631] [Q&A] [answered]
apologies if i'm just not seeing it, but is there a good overview of how to extend ComfyUI somewhere?

custom nodes seem pretty easy to make and i think i got that down. but if i want a new widget, where does that code go? `web/extensions/` isn't in `.gitignore` so i don't think i'm supposed to just plop code there, and ideally my node's python and js would be together with the whole thing cloned under `custom_nodes`.

doesn't look like `GET /extensions` crawls `custom_nodes/` at all... should it? or is there something i'm missing?

**Comment by comfyanonymous** [comment, marked as answer]
There are two type of extensions, frontend extensions which go in web/extension folder and custom nodes which go in the custom_nodes folder.

This is very likely going to be organized a bit better in the future but this is how it works now.

**Comment by hgw405025254** [comment]
ComfyUI-Mediapayer存在吗

**Reply by ltdrdata** [reply]
There is no such custom node. However, if it's simply for video processing, you can search for "video" in ComfyUI-Manager to find various custom nodes.
